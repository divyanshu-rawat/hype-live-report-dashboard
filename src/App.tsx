import React, { useEffect, useState } from 'react';

import styled from '@emotion/styled';

import { EventView } from './components/event-view';
import { EventsAggregateView } from './components/events-aggregate-view';
import { GlobalStyles } from './theme/global';
import { PrimaryTitle } from './theme/typography';
import { eventType } from './types/global';
import { fetchData } from './utils/index';

const AppContainer = styled.div`
  max-width: 1200px;
  margin: 0 auto;
`;

const Title = styled(PrimaryTitle)`
  text-align: center; 
  font-weight: 600;
`;

export const App: React.FC = () => {
  const [events, setEvents] = useState<eventType[]>([]);
  useEffect(() => {
    fetchData().then((response) => setEvents(response));
  }, [events]);
  return (
    <AppContainer>
      <GlobalStyles />
      <Title>Hype Live Report</Title>
      {events.length > 0 && <EventsAggregateView events={events} />}
      {events.length > 0 && <EventView events={events} />}
    </AppContainer>
  );
};


