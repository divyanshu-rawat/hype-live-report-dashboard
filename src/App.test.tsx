import React from 'react';

import { cleanup, render } from '@testing-library/react';

import { App } from './App';

afterEach(cleanup);

test('renders App', () => {
  render(<App />);
});

it('should take a App snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
