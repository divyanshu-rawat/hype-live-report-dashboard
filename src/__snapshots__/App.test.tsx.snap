// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a App snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-ee25lq"
  >
    <h2
      class="css-ty6nqg"
      data-testid="AppTitle"
    >
      Hype Live Report
    </h2>
  </div>
</DocumentFragment>
`;
